### ISSUE TITLE
[BUG] Failure to build master branch due to a failling test 

### ISSUE THREAD

**Issue by cvgaviao** [OpenAPITools/openapi-generator#2551]
I have cloned the openapi-generator today in order to test it with a snapshot version of swagger-parser containing a patch to an issue that bitten my spec file: https://github.com/swagger-api/swagger-parser/issues/1063

So, I have changed the swagger-parser version in the master pom and tried to build.

But I'm getting a failure in this test class: `org.openapitools.codegen.java.JavaModelTest` on both maven or eclipse testng run:
```

org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 1, Warning count: 0
Errors: 
	-attribute paths.'/pet/findByStatus'(get).parameters.[status].schemas.default is not of type `array`

	at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:615)
	at org.openapitools.codegen.java.JavaModelTest.generateModel(JavaModelTest.java:1230)

```

**Comment by cvgaviao** [comment]
Did some investigation in the failing method: generateModel().

Interesting to note that other tests are using `petstore.yarn` and no one is failing, but this test is using `src/test/resources/3_0/petstore.json`

Then I've converted this json into yaml and compared with the petstore.yarn in the master branch.

As I was suspecting the `src/test/resources/3_0/petstore.json` has a structure problem.

**Comment by cvgaviao** [comment]
I successfully generated a code from my spec now, but in order to run the generator with latest swagger-parser was necessary to include explicitly this dependency in `openapi-generator/modules/openapi-generator/pom.xml`:
```
        <dependency>
            <groupId>io.swagger.parser.v3</groupId>
            <artifactId>swagger-parser</artifactId>
            <version>${swagger-parser-version}</version>
        </dependency>
```

**Comment by wing328** [comment]
We just released 4.0.0-beta3. Can you try that when you've time? If it's still an issue, please let us know (closing the issue for the time being)
